projects/graph/src/graph.py

```diff
@@ -1,9 +1,194 @@
+import random
 """
 Simple graph implementation compatible with BokehGraph class.
 """
 
+class Queue():
+	def __init__(self):
+		self.queue = []
+	def enqueue(self, value):
+		self.queue.append(value)
+	def dequeue(self):
+		if self.size() > 0:
+			return self.queue.pop(0)
+		else:
+			return None
+	def size(self):
+		return len(self.queue)
 
+class Stack():
+	def __init__(self):
+		self.stack = []
+	def push(self, value):
+		self.stack.append(value)
+	def pop(self):
+		if self.size() > 0:
+			return self.stack.pop()
+		else:
+			return None
+	def size(self):
+		return len(self.stack)
+
+class Vertex:
+	def __init__(self, vertex_id, x=None, y=None, value=None, color="white"):
+		self.id = int(vertex_id)
+		self.x = x
+		self.y = y
+		self.value = value
+		self.color = color
+		self.edges = set()
+		if self.x is None:
+			self.x = 2 * (self.id // 3) + self.id / 10 * (self.id % 3)
+		if self.y is None:
+			self.y = 2 * (self.id % 3) + self.id / 10 * (self.id // 3)
+		if self.value is None:
+			self.value = self.id
+
+	def __str__(self):
+		return f'edges: {self.edges}'
+
+"""Represent a graph as a dictionary of vertices mapping labels to edges."""
 class Graph:
-    """Represent a graph as a dictionary of vertices mapping labels to edges."""
-    def __init__(self):
-        pass  # TODO
+	def __init__(self):
+		self.vertices = {}
+	
+	def add_vertex(self, vertex_id):
+		self.vertices[vertex_id] = Vertex(vertex_id)
+
+	def add_edge(self, v1, v2, bi):
+		check_one = False
+		check_two = False
+
+		for i in self.vertices: 
```

Comment:
This is interesting, I think there may be more concise ways to write these lines, but they're very lucid and understandable, and reusable. Nice.